Regularized_Logistic_Regression.py

```diff
@@ -0,0 +1,136 @@
+        self.gradients = {'derivative_weights' : derivative_weights,
+                    'derivative_bias' : derivative_bias}
+        return cost
+
+    # updating our parameters {weights and bias}
+    def update_weights(self, X_train, y_train):
+        cost_list = []
+        for i in range(self.epochs):
+        	# computing the forward propagation
+            cost = self.forwardPropagation(X_train, y_train)
+            cost_list.append(cost)
+
+            # Here we are updating our theta (weights) and theta0 (bias values)
+            self.theta = (self.theta * (1 - (self.alpha * self.lambda_)/self.m))  - (self.alpha * self.gradients['derivative_weights'] + ((self.lambda_) * np.sum(self.theta))/self.m)
+            self.theta0 = self.theta0 - self.alpha * self.gradients['derivative_bias']
+
+            if i % 10000 == 0:
+            	print('Epochs ==>> ',i)
+
+        self.parameters = {'weight': self.theta, 'bias' : self.theta0}
+        plt.plot(range(len(cost_list)), cost_list)
+        plt.xlabel('Number of Iterations')
+        plt.ylabel('Cost function J(theta)')
+        plt.show()
+
+    # Training the model
+    def fit(self, X_train, y_train):
+        X_train = X_train.T
+        y_train = y_train.T
+        self.m = X_train.shape[0]
+        self.n = X_train.shape[1]
+        self.initialize_theta(self.m) # randomly initialize our theta values
+        X_train = self.minMaxNormalization(X_train) # normalizing our input features
+        self.update_weights(X_train, y_train) # updating our weights parameters 
+
+    # Make predictions for us
+    def predict(self, X_test):
+        self.theta = self.parameters['weight'] # optimium weight
+        self.theta0 = self.parameters['bias'] # optimum bias
+        X_test = X_test.T
+        z = self.sigmoid(np.dot(self.theta.T, X_test) + self.theta0)
+        y_prediction = np.zeros((1, X_test.shape[1])) # number of samples
+        for i in range(z.shape[1]):
+            if z[0, i] < 0.5:
+                y_prediction[0, i] = 0
+            else:
+                y_prediction[0, i] = 1
+        return y_prediction[0]
+
+
+# loading the data
+breast_cancer = load_breast_cancer()
+
+# intializing our feature matrix and target vectors
+X = breast_cancer.data
+y = breast_cancer.target
+
+# checking the shape of our models
+print(np.shape(X))
+print(np.shape(y))
+
+# spliting our data into train and test sets
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, shuffle=True)
+
+# checking the shape to validate what was done
+print('Checking the shapes...')
+print(X_train.shape)
+print(X_test.shape)
+print(y_train.shape)
+print(y_test.shape)
+print('')
+
+# calling our logistic regression method. I also integrated the regularization feature to help
+# prevent our model from overfitting. So penalties are been put on our theta (weight) values
+model = LogisticRegression(epochs=100000, lambda_=0.001, alpha=0.01)
+
```

Comment:
You can also print out the weights and biases here.

By typing

print(model.parameters['weight'])
print(model.parameters['bias])